src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -1057,26 +1057,24 @@ static void footpath_remove_tile_element(TileElement* tileElement)
 static TileElement* footpath_get_tile_element_to_remove()
 {
     TileElement* tileElement;
-    int32_t x, y, z, zLow;
+    int32_t z, zLow;
 
-    x = gFootpathConstructFromPosition.x / 32;
-    y = gFootpathConstructFromPosition.y / 32;
-    if (x >= 256 || y >= 256)
+    if (map_is_location_valid(gFootpathConstructFromPosition))
     {
         return nullptr;
     }
 
-    z = (gFootpathConstructFromPosition.z >> 3) & 0xFF;
-    zLow = z - 2;
+    z = gFootpathConstructFromPosition.z & std::numeric_limits<int32_t>::max();
```

Comment:
This is not the same. There has to be a reason it maxes out at 255, and you just raised that limit enormously.

Reply:
Yeah it should be something like `z = std::min(255 * COORDS_Z_STEP, gFootpathConstructFromPosition.z)`

Reply:
Somehow I thought it would be equivalent, I'll update